lib/puppet/type/cloudwatch_alarm.rb

```diff
@@ -0,0 +1,91 @@
+Puppet::Type.newtype(:cloudwatch_alarm) do
+  @doc = 'Type representing an AWS CloudWatch Alarm.'
+
+  ensurable
+
+  newparam(:name, namevar: true) do
+    desc 'The name of the alarm.'
+    validate do |value|
+      fail 'alarms must have a name' if value == ''
+    end
+  end
+
+  newproperty(:metric) do
+    desc 'The name of the metric to track.'
+    validate do |value|
+      fail 'metric must not be blank' if value == ''
+    end
+  end
+
+  newproperty(:namespace) do
+    desc 'The namespace of the metric to track.'
+    validate do |value|
+      fail 'namespace must not be blank' if value == ''
+    end
+  end
+
+  newproperty(:statistic) do
+    desc 'The statistic to track for the metric.'
+    validate do |value|
+      fail 'statistic must not be blank' if value == ''
+    end
+  end
+
+  newproperty(:period) do
+    desc 'The periodicity of the alarm check.'
+    validate do |value|
+      fail 'period cannot be blank' if value == ''
+    end
+    munge do |value|
+      value.to_i
+    end
+  end
+
+  newproperty(:evaluation_periods) do
+    desc 'The number of checks to use to confirm the alarm.'
+    validate do |value|
+      fail 'evaluation periods cannot be blank' if value == ''
+    end
+    munge do |value|
+      value.to_i
+    end
+  end
+
+  newproperty(:threshold) do
+    desc 'The threshold used to trigger the alarm.'
+    validate do |value|
+      fail 'threshold cannot be blank' if value == ''
+    end
+    munge do |value|
+      value.to_f
+    end
+  end
+
+  newproperty(:comparison_operator) do
+    desc 'The operator to use to test the metric.'
+    validate do |value|
+      fail 'comparison operator must not be blank' if value == ''
+    end
+  end
+
+  newproperty(:region) do
+    desc 'The region in which to launch the instances.'
+    validate do |value|
+      fail 'region should not contain spaces' if value =~ /\s/
+      fail 'region should not be blank' if value == ''
+    end
+  end
+
+  newparam(:dimensions, :array_matching => :all) do
+    desc 'The dimensions to filter the alerm by.'
```

Comment:
Should these parameters have validation as well? It looks like https://github.com/puppetlabs/puppetlabs-aws/pull/30/files#diff-84de254bad7394aa4014979a5b0fde38R70 requires this this to be a hash, the example uses an array, and it (predictably) messes things up when I try to use a string.